oslo_cache/core.py

```diff
@@ -161,12 +161,15 @@ def _build_cache_config(conf):
     # does not need the explicit flush_on_reconnect
     conf_dict.setdefault('%s.arguments.url' % prefix,
                          conf.cache.memcache_servers)
-    for arg in ('dead_retry', 'socket_timeout', 'pool_maxsize',
-                'pool_unused_timeout', 'pool_connection_get_timeout',
-                'pool_flush_on_reconnect', 'sasl_enabled', 'username',
-                'password'):
-        value = getattr(conf.cache, 'memcache_' + arg)
-        conf_dict['%s.arguments.%s' % (prefix, arg)] = value
+
+    # This removes username and password that are needed for redis_sentinel backend
+    if conf.cache.backend != "oslo_cache.redis_sentinel":
+        for arg in ('dead_retry', 'socket_timeout', 'pool_maxsize',
+                    'pool_unused_timeout', 'pool_connection_get_timeout',
+                    'pool_flush_on_reconnect', 'sasl_enabled', 'username',
+                    'password'):
+            value = getattr(conf.cache, 'memcache_' + arg)
+            conf_dict['%s.arguments.%s' % (prefix, arg)] = value
```

Comment:
:thinking: This may be an area where upstream will request the change. It might be good to include line 162 as well. But I have to this about this bit more if there is a nicer way to achieve this :thinking: